.gitmodules

```diff
@@ -1,36 +1,100 @@
 [submodule "libsodium/libsodium"]
 	path = libsodium/libsodium
 	url = https://github.com/jedisct1/libsodium.git
+	# libsodium doesn't tag new releases since 0.18.0, 'stable' branch is considered
+	# to be the latest stable release. See:
+	# https://github.com/jedisct1/libsodium/issues/1041#issuecomment-799812417
+	# However since idf_component.yml had the version already set to 0.20.0,
+	# we can't update the version automatically (it would be set it back to 0.18.0~n)
+	# autoupdate = false
 [submodule "cbor/tinycbor"]
 	path = cbor/tinycbor
 	url = https://github.com/intel/tinycbor.git
+	autoupdate = true
+	autoupdate-branch = main
+	autoupdate-tag-glob = v[0-9]*.[0-9]*.[0-9]*
+	autoupdate-manifest = cbor/idf_component.yml
+	autoupdate-ver-regex = v([0-9]+).([0-9]+).([0-9]+)
 [submodule "nghttp/nghttp2"]
 	path = nghttp/nghttp2
 	url = https://github.com/nghttp2/nghttp2.git
+	autoupdate = true
+	autoupdate-branch = master
+	autoupdate-tag-glob = v[0-9]*.[0-9]*.[0-9]*
+	autoupdate-manifest = nghttp/idf_component.yml
+	autoupdate-ver-regex = v([0-9]+).([0-9]+).([0-9]+)
 [submodule "expat/expat"]
 	path = expat/expat
 	url = https://github.com/libexpat/libexpat.git
+	autoupdate = true
+	autoupdate-branch = master
+	autoupdate-tag-glob = R_[0-9]*_[0-9]*_[0-9]*
+	autoupdate-manifest = expat/idf_component.yml
+	autoupdate-ver-regex = R_([0-9]+)_([0-9]+)_([0-9]+)
 [submodule "coap/libcoap"]
 	path = coap/libcoap
 	url = https://github.com/obgm/libcoap.git
+	autoupdate = true
+	autoupdate-branch = develop
+	autoupdate-tag-glob = v[0-9]*.[0-9]*.[0-9]*
+	autoupdate-include-lightweight = true
+	autoupdate-manifest = coap/idf_component.yml
+	autoupdate-ver-regex = v([0-9]+).([0-9]+).([0-9]+)
 [submodule "usb/usb_host_uvc/libuvc"]
 	path = usb/usb_host_uvc/libuvc
 	url = https://github.com/libuvc/libuvc.git
+	autoupdate = true
+	autoupdate-branch = master
+	autoupdate-tag-glob = v[0-9]*.[0-9]*.[0-9]*
+	autoupdate-include-lightweight = true
+	# Not updating the manifest automatically,
+	# since the component version is ahead of the upstream version
 [submodule "eigen/eigen"]
 	path = eigen/eigen
 	url = https://gitlab.com/libeigen/eigen.git
+	autoupdate = true
+	autoupdate-branch = 3.4
+	autoupdate-tag-glob = [0-9]*.[0-9]*.[0-9]*
+	autoupdate-include-lightweight = true
+	autoupdate-manifest = eigen/idf_component.yml
+	autoupdate-ver-regex = ([0-9]+).([0-9]+).([0-9]+)
 [submodule "fmt/fmt"]
 	path = fmt/fmt
 	url = https://github.com/fmtlib/fmt.git
+	autoupdate = true
+	autoupdate-branch = master
+	autoupdate-tag-glob = [0-9]*.[0-9]*.[0-9]*
+	autoupdate-include-lightweight = true
+	autoupdate-manifest = fmt/idf_component.yml
+	autoupdate-ver-regex = ([0-9]+).([0-9]+).([0-9]+)
 [submodule "esp_delta_ota/detools"]
 	path = esp_delta_ota/detools
 	url = https://github.com/eerimoq/detools.git
+	# not updating automatically, as this is not an upstream repo
```

Comment:
We do use source from this repository in our component build. We should add this component in the auto update list.

Reply:
Yes, they are not related. Our component follows its own versioning policy here. We can handle the update for this case on-demand basis.